File: frontend/src/types/symbol.ts
export interface SymbolSearchResult {
  symbol: string;
  brsymbol: string;
  name: string | null;
  exchange: string;
  brexchange: string | null;
  token: string | null;
  expiry: string | null;
  strike: number | null;
  lotsize: number | null;
  instrumenttype: string | null;
  tick_size: number | null;
}

export const EXCHANGES: ReadonlyArray<{ value: string; label: string }> = [
  { value: "NSE", label: "NSE — Equity" },
  { value: "NFO", label: "NFO — NSE Futures & Options" },
  { value: "BSE", label: "BSE — Equity" },
  { value: "BFO", label: "BFO — BSE Futures & Options" },
  { value: "MCX", label: "MCX — Commodities" },
  { value: "CDS", label: "CDS — NSE Currency" },
  { value: "BCD", label: "BCD — BSE Currency" },
  { value: "NSE_INDEX", label: "NSE_INDEX — NSE Indices" },
  { value: "BSE_INDEX", label: "BSE_INDEX — BSE Indices" },
  { value: "GLOBAL_INDEX", label: "GLOBAL_INDEX — World Indices" },
  { value: "GLOBAL_INDICATOR", label: "GLOBAL_INDICATOR — FX & Commodity Refs" },
];

File: backend/models/symbol.py
from sqlalchemy import Column, Integer, String, Float, Index

from backend.database import Base


class SymToken(Base):
    __tablename__ = "symtoken"

    id = Column(Integer, primary_key=True)
    symbol = Column(String(100), nullable=False, index=True)
    brsymbol = Column(String(100), nullable=False, index=True)
    name = Column(String(100), nullable=True)
    exchange = Column(String(20), nullable=False, index=True)
    brexchange = Column(String(20), nullable=True)
    token = Column(String(50), nullable=True, index=True)
    expiry = Column(String(20), nullable=True)
    strike = Column(Float, nullable=True)
    lotsize = Column(Integer, nullable=True)
    instrumenttype = Column(String(20), nullable=True)
    tick_size = Column(Float, nullable=True)

    __table_args__ = (
        Index("idx_symtoken_symbol_exchange", "symbol", "exchange"),
        Index("idx_symtoken_brsymbol_exchange", "brsymbol", "exchange"),
    )
